### ISSUE TITLE
How to set different retention for each topic?

### ISSUE THREAD

**Issue by rc1cent** [confluentinc/confluent-kafka-dotnet#861]
Description
===========
Hi.
I'm using `auto.create.topics.enable=true` option to create new topics. I want to set retention for newly created topics but I want to set different retention for each topic . I've tried to setup default configuration for topics like this:
```csharp
var config = new Dictionary<string, object>
{
    { "default.topic.config", new Dictionary<string, object>()
        ...
        {
            ...
            { "retention.ms", TimeSpan.FromDays(3).TotalMilliseconds }
        }
    }
}
```
but I'm getting an exception:
```
Message: System.InvalidOperationException : No such configuration property: "retention.ms"
StackTrace:	
   at Confluent.Kafka.Impl.SafeConfigHandle.Set(String name, String value)
   at Confluent.Kafka.Producer.<>c__DisplayClass23_0.<.ctor>b__3(KeyValuePair`2 kvp)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Confluent.Kafka.Producer..ctor(IEnumerable`1 config, Boolean manualPoll, Boolean disableDeliveryReports)
   at Confluent.Kafka.Producer`2..ctor(IEnumerable`1 config, ISerializer`1 keySerializer, ISerializer`1 valueSerializer, Boolean manualPoll, Boolean disableDeliveryReports)
   at Confluent.Kafka.Producer`2..ctor(IEnumerable`1 config, ISerializer`1 keySerializer, ISerializer`1 valueSerializer)
   ...
```
What am I doing wrong? Am I using wrong option and/or in wrong way?
In situation that I have I can't use scripts from command line so the only one option is to use Confluent.Kafka.0.11.6 package.

I'm using:
- Confluent.Kafka.0.11.6 NuGet package.
- kafka_2.11-2.1.0.tgz Kafka package
- Windows 10

**Comment by mhowlett** [comment]
It's generally a good idea to disable auto topic creation as it can lead to unexpected problems (*any* broker request that references the topic will cause it to be created).

You can achieve what you want using `AdminClient.CreateTopicsAsync`. This is available in 1.0.0-RC3. 1.0 final will be released very shortly (RC3 likely is the final version, but we're just letting it sit a couple more days before releasing).

**Comment by pranavrth** [comment]
Closing the issue as it is available now.
